tests/fuser_mqubit_test.cc

```diff
@@ -371,6 +373,269 @@ TEST(FuserMultiQubitTest, Simulation) {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(0, {2, 3}),
+      CreateDummyGate(0, {4, 5}),
+      CreateDummyGate(1, {1, 2}),
+      CreateDummyGate(1, {3, 4}),
+      CreateDummyGate(1, {5, 0}),
+    };
+
+    param.max_fused_size = 3;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+    EXPECT_EQ(fused_gates.size(), 4);
+  }
+
+  {
+    unsigned num_qubits = 8;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(0, {2, 3}),
+      CreateDummyGate(0, {4, 5}),
+      CreateDummyGate(0, {6, 7}),
+      CreateDummyGate(1, {1, 2}),
+      CreateDummyGate(1, {3, 4}),
+      CreateDummyGate(1, {5, 6}),
+      CreateDummyGate(1, {7, 0}),
+    };
+
+    param.max_fused_size = 5;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+    EXPECT_EQ(fused_gates.size(), 3);
+  }
+
+  {
+    unsigned num_qubits = 3;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {1, 2}),
+      CreateDummyGate(1, {0, 1}),
+      CreateDummyGate(2, {1, 2}),
+    };
+
+    param.max_fused_size = 3;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+    EXPECT_EQ(fused_gates.size(), 1);
+  }
+
+  {
+    unsigned num_qubits = 3;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(1, {0, 2}),
+      CreateDummyGate(2, {1, 2}),
+    };
+
+    param.max_fused_size = 3;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+    EXPECT_EQ(fused_gates.size(), 1);
+  }
+
+  {
+    unsigned num_qubits = 6;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {0, 1}),
+      CreateDummyGate(1, {2, 3}),
+      CreateDummyGate(2, {4, 5}),
+      CreateDummyGate(3, {1, 2}),
+      CreateDummyGate(4, {2, 3, 4}),
+    };
+
+    param.max_fused_size = 3;
+    auto fused_gates = Fuser::FuseGates(
+        param, num_qubits, circuit.begin(), circuit.end());
+
+    EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
+    EXPECT_EQ(fused_gates.size(), 4);
+  }
+
+  {
+    unsigned num_qubits = 7;
+    std::vector<DummyGate> circuit = {
+      CreateDummyGate(0, {1, 6}),
+      CreateDummyGate(1, {0, 1}),
+      CreateDummyGate(1, {4, 5}),
+      CreateDummyGate(2, {1, 2, 3}),
+      CreateDummyGate(2, {5, 6}),
+      CreateDummyGate(3, {3, 4}),
+    };
+
+    {
+      param.max_fused_size = 3;
+      auto fused_gates = Fuser::FuseGates(
+          param, num_qubits, circuit.begin(), circuit.end());
+
+      EXPECT_TRUE(TestFusedGates(num_qubits, circuit, fused_gates));
```

Comment:
No expectation on `fused_gates.size()` here?

Reply:
Yes, the reason here and below is that the current version of the fuser does not give the optimal number of fused gates. Perhaps some future version will be better.